mlflow/pyfunc/__init__.py

```diff
@@ -656,15 +674,285 @@ def _warn_dependency_requirement_mismatches(model_path):
                 f"Python environment:\n{mismatch_str}"
             )
             _logger.warning(warning_msg)
+            return False
+        else:
+            return True
     except Exception as e:
         _logger.warning(
             f"Encountered an unexpected error ({repr(e)}) while detecting model dependency "
             "mismatches. Set logging level to DEBUG to see the full traceback."
         )
         _logger.debug("", exc_info=True)
+        return e
+
+
+_PYENV_CMD = 'pyenv'
+_VIRTUALENV_CMD = 'virtualenv'
+
+_DATABRICKS_NFS_CACHE_ROOT_DIR = "/local_disk0/.ephemeral_nfs/mlflow/cache"
+
+
+_PICKLE_PROTOCOL_FOR_RESTORE_PY_ENV = 3
```

Comment:
Do we want to stick with Protocol 4 (supported since Py3.4) which is the default in Py3.8+ to support, as the docs (https://docs.python.org/3/library/pickle.html#data-stream-format) mention "adds support for very large objects, pickling more types of objects, and some data format optimizations"?

Reply:
I think we can automatically generate best Protocol number, according to the virtual env python version.